### PR TITLE
[kube-stack] Bump opentelemetry-operator chart to 0.71

### PR DESCRIPTION
**Pull request by rogercoll**
Generated examples with: `make generate-examples CHARTS=opentelemetry-kube-stack`

This will fix an issue when using the Helm Chart with a collector's image > 0.110.0:

```
$ k logs -n opentelemetry-operator-system opentelemetry-kube-stack-daemon-collector-xggww                          
invalid argument "-component.UseLocalHostAsDefaultHost" for "--feature-gates" flag: feature gate "component.UseLocalHostAsDefaultHost" is stable, can not be disabled
```

### CODE REVIEW

**Review by gizas — APPROVED** [review]

**Comment by rogercoll** [comment]
Example configuration:

```yaml
defaultCRConfig:
  image:
    repository: "otel/opentelemetry-collector-k8s"
    tag: "0.110.0"
```

**Comment by rogercoll** [comment]
Feature gate was removed in Operator's v0.110.0: https://github.com/open-telemetry/opentelemetry-operator/pull/3336/files#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4edR28

**Review by swiatekm — APPROVED** [review]

File: charts/opentelemetry-kube-stack/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v2
 name: opentelemetry-kube-stack
-version: 0.2.3
+version: 0.2.4
```
Comment (jaronoff97) [inline charts/opentelemetry-kube-stack/Chart.yaml:3]: we can change this to 0.3.0

File: charts/opentelemetry-kube-stack/Chart.yaml [hunk 2]
```diff
@@ -23,7 +23,7 @@ dependencies:
     condition: crds.install
```
Comment (jaronoff97) [inline charts/opentelemetry-kube-stack/Chart.yaml:23]: can you also update the CRDs too please?

**Review by jaronoff97 — COMMENTED** [review]
two minor things, thank you!

File: charts/opentelemetry-kube-stack/Chart.yaml (hunk 1 above)
Comment (rogercoll) [reply charts/opentelemetry-kube-stack/Chart.yaml]: Definitely, changed in https://github.com/open-telemetry/opentelemetry-helm-charts/pull/1384/commits/d6c39c1e7107751d2dc581226b6ea7ac57166db4

File: charts/opentelemetry-kube-stack/Chart.yaml (hunk 2 above)
Comment (rogercoll) [reply charts/opentelemetry-kube-stack/Chart.yaml:23]: https://github.com/open-telemetry/opentelemetry-helm-charts/pull/1384/commits/d6c39c1e7107751d2dc581226b6ea7ac57166db4

thanks for the feedback!

**Comment by rogercoll** [comment]
@jaronoff97 I had to bump the Python version used in the CI as it seems 3.7 was removed from the used runners:

```
  Version 3.7 was not found in the local cache
  Error: The version '3.7' with architecture 'x64' was not found for Ubuntu 24.04.
  The list of all available versions can be found here: https://raw.githubusercontent.com/actions/python-versions/main/versions-manifest.json

```

**Review by jaronoff97 — APPROVED** [review]

**Review by TylerHelmuth — APPROVED** [review]

**Review by jaronoff97 — APPROVED** [review]

**Pull request merged**